main_test.go

```diff
@@ -50,3 +50,16 @@ func getEnv(key, fallback string) string {
 	}
 	return fallback
 }
+
+func TestIsAllowedZones(t *testing.T) {
```

Comment:
With the feedback above, I think we should add two more cases:

1. Exact match on domain (e.g., `example.com.`)
2. Non-match for suffix that isn't a subdomain (e..g, `notexample.com.`)